File: yejipractice/PRG/정렬/가장_큰_수.py
numbers = [3, 30, 34, 5, 9]


def solution(numbers):
    numbers = list(map(str, numbers))
    numbers.sort(key=lambda x: x*3, reverse=True)
    return str(int("".join(numbers)))


print(solution(numbers))

# num의 인수값이 1000 이하이므로 3자리수로 맞춘 뒤 비교
# [0, 0, 0]일 경우 000이 아닌 0 반환하도록

File: jifrozen/정렬/가장_큰_수.java
package 정렬;
import java.util.*;
class 가장_큰_수 {
	public String solution(int[] numbers) {
		String answer = "";
		String[] arr=new String[numbers.length];
		for(int i=0;i<numbers.length;i++){
			arr[i]=numbers[i]+"";
		}
		Arrays.sort(arr,new Comparator<String>(){
			@Override
			public int compare(String n1,String n2){
				return (n2+n1).compareTo(n1+n2);
			}
		});
		for(int i=0;i<numbers.length;i++){
			answer+=arr[i];
		}

		if(answer.charAt(0)=='0'){
			return "0";
		}
		return answer;
	}
}